algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -448,6 +450,50 @@ case class CMSItem[K: Ordering](item: K, override val params: CMSParams[K]) exte
 }
 
 /**
+ * Used for holding a small set of elements.
+ */
+case class CMSItems[K: Ordering](items: List[K], override val totalCount: Long, override val params: CMSParams[K]) extends CMS[K](params) {
+
+  override def +(x: K, count: Long): CMS[K] = CMSItems(x :: items, totalCount + 1, params).toCMS
+
+  override def ++(other: CMS[K]): CMS[K] = {
+    other match {
+      case other: CMSZero[_] => this
+      case other: CMSItem[K] => this + other.item
+      case other: CMSItems[K] => {
+        val (larger, smaller) =
+          if (totalCount > other.totalCount)
+            (this, other)
+          else
+            (other, this)
+
+        CMSItems(
+          larger.items.reverse ::: (smaller.items),
```

Comment:
1. don't we want the method `reverse_:::` which combines these two to save cost?
2. don't we want the smaller one on the left so we push the fewer items onto the bigger?